src/MeetingsSDKAdapter.js

```diff
@@ -509,6 +590,157 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
     return concat(getDisplayData$, localMediaUpdateEvent$);
   }
 
+  /**
+   * Attempts to start/stop sharing of the given meeting ID.
+   * If success, a sharing start/stop event is dispatched.
+   *
+   * @param {string} ID ID of the meeting to start/stop sharing
+   * @memberof MeetingJSONAdapter
+   */
+  async handleLocalShare(ID) {
+    const sdkMeeting = this.fetchMeeting(ID);
+
+    if (!sdkMeeting) {
+      // eslint-disable-next-line no-console
+      console.error(`
+        Unable to update share for meeting "${ID}": sdk meeting does not exist
+      `);
+
+      return;
+    }
+
+    if (!sdkMeeting.canUpdateMedia()) {
+      // eslint-disable-next-line no-console
+      console.error(`
+      Unable to update share for meeting "${ID}": cannot update media due to SDP negotiation in progress
+    `);
+
+      return;
+    }
+
+    const enableSharingStream = async () => {
+      const response = await sdkMeeting.getMediaStreams({sendShare: true});
+
+      // eslint-disable-next-line prefer-destructuring
+      this.meetings[ID].localShare = response[1];
+
+      // Due to SDK limitation around local media updates,
+      // we need to emit a custom event for video mute updates
+      sdkMeeting.emit(EVENT_MEDIA_LOCAL_UPDATE, {
+        control: SHARE_CONTROL,
+        state: MeetingControlState.DISABLED,
+      });
+
+      await sdkMeeting.updateShare({stream: response[1], sendShare: true, receiveShare: true});
+    };
+
+    const disableSharingStream = async () => {
+      sdkMeeting.emit(EVENT_MEDIA_LOCAL_UPDATE, {
```

Comment:
I think you should emit only after the share stream was successfully updated and stopped, instead of the beginning

Reply:
This is a workaround for a bug of async updateShare() call. The async call may need ~30s to finish execution, or even cannot finish execution. If the emit is called after the async call, the user will never see the state change of UI and get confused. So we moved this emit call to before the async call.

Reply:
We should probably mark it with a TODO and a jira

Reply:
Reported as this jira: https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-148910

Reply:
Thanks for reporting it 👍🏻